[Osquery] Add docs for Osquery API (#137162)
elastic/kibana: docs/api/osquery-manager/packs/get-all.asciidoc

[[osquery-manager-packs-api-get-all]]
=== Get packs API
++++
<titleabbrev>Get packs</titleabbrev>
++++

experimental[] Get packs.

[[osquery-manager-packs-api-get-all-request]]
==== Request

`GET <kibana host>:<port>/api/osquery/packs`

`GET <kibana host>:<port>/s/<space_id>/api/osquery/packs`

[[osquery-manager-packs-api-get-all-params]]
==== Path parameters

`space_id`::

> (Optional, string) An identifier for the space. If `space_id` is not provided in the URL, the default space is used.

Reviewer: ```suggestion
(Optional, string) The space identifier. When `space_id` is not provided in the URL, the default space is used.
```

=== Query parameters

`page`::

> (Optional, integer) The page number to return. Defaults to `1`.

Reviewer: ```suggestion
(Optional, integer) The page number to return. The default is `1`.
```

`perPage`::

> (Optional, integer) The number of rules to return per page. Defaults to `20`.

Reviewer: ```suggestion
(Optional, integer) The number of rules to return per page. The default is `20`.
```

`sortField`::

> (Optional, string) Determines which field is used to sort the results,

Reviewer: ```suggestion
(Optional, string) Specifies the field that sorts the results. Options include `createdAt` or `updatedAt`.
```

> `createdAt` or `updatedAt`. Defaults to `createdAt`.

Reviewer: ```suggestion
The default is `createdAt`.
```

+

> NOTE: Even though the JSON case object uses `created_at` and `updated_at`

Reviewer: ```suggestion
NOTE: Even though the JSON case object uses the `created_at` and `updated_at`
```

fields, you must use `createdAt` and `updatedAt` fields in the URL
query.

`sortOrder`::

> (Optional, string) Determines the sort order, which can be `desc` or `asc`.

Reviewer: ```suggestion
(Optional, string) Specifies the sort order. Options include `desc` or `asc`.
```

> Defaults to `desc`.

Reviewer: ```suggestion
The default is `desc`.
```

[[osquery-manager-packs-api-get-all-codes]]
==== Response code

`200`::
Indicates a successful call.

[[osquery-manager-packs-api-get-all-example]]
==== Example

> Retrieve first 10 packs:

Reviewer: ```suggestion
Retrieve the first 10 packs:
```

[source,sh]
--------------------------------------------------
$ curl -X GET api/osquery/packs?page=1&perPage=10&sortField=updatedAt&sortOrder=asc
--------------------------------------------------
// KIBANA

> The API returns a JSON object listing the retrieved packs. For example:

Reviewer: ```suggestion
The API returns a JSON object with the retrieved packs:
```
